apps/backend/src/modules/groups/group.repository.ts

```diff
@@ -45,12 +46,14 @@ class GroupRepository implements Repository {
 		return Promise.resolve(null);
 	}
 
-	public async findAll(): Promise<GroupEntity[]> {
+	public async findAll(): Promise<GroupGetAllResponseDto> {
 		const groups = await this.groupModel
 			.query()
 			.withGraphFetched("[permissions, users]");
 
-		return groups.map((group) => GroupEntity.initialize(group));
+		return {
+			items: groups.map((group) => GroupEntity.initialize(group).toObject()),
```

Comment:
remove .toObject call on repository level, we need to return entity
```suggestion
			items: groups.map((group) => GroupEntity.initialize(group)),
```